lib/model/basket_item.rb

```diff
@@ -0,0 +1,34 @@
+module Shop
+  class BasketItem
+    attr_reader :product_id, :quantity
+
+    ProductNotFound = Class.new(StandardError)
+    WrongQuantityRequested = Class.new(StandardError)
+    NoSuchItem = Class.new(StandardError)
+
+    def initialize(product_id:, quantity: 1)
+      @product_id = product_id
+      @quantity = quantity
+
+      raise ArgumentError unless quantity.is_a?(Integer)
+      raise ArgumentError unless quantity > 0
+    end
+
+    def fetch_product
+      FetchProduct.new.call(product_id)
+    end
+
+    def unit_price
+      product = fetch_product
```

Comment:
I think you could simply use `FetchProduct.new.call(product_id)` here and get rid of separate method. This would lead to less code and would still be readable imo.